src/phoenix/server/api/queries.py

```diff
@@ -187,44 +186,33 @@ async def node(self, id: GlobalID, info: Info[Context, None]) -> Node:
                 metadata=dataset.metadata_,
             )
         elif type_name == "DatasetExample":
-            example_rowid = node_id
-            revision_rowid = (
+            example_id = node_id
+            latest_revision_id = (
                 select(func.max(models.DatasetExampleRevision.id))
-                .where(models.DatasetExampleRevision.dataset_example_id == example_rowid)
+                .where(models.DatasetExampleRevision.dataset_example_id == example_id)
                 .scalar_subquery()
             )
             async with info.context.db() as session:
-                example_result = (
-                    await session.execute(
-                        select(models.DatasetExample, models.DatasetExampleRevision)
-                        .select_from(models.DatasetExampleRevision)
-                        .join(
-                            models.DatasetExample,
-                            onclause=models.DatasetExampleRevision.dataset_example_id
-                            == models.DatasetExample.id,
-                        )
-                        .where(
-                            and_(
-                                models.DatasetExample.id == example_rowid,
-                                models.DatasetExampleRevision.id == revision_rowid,
-                                models.DatasetExampleRevision.revision_kind != "DELETE",
-                            )
+                example = await session.scalar(
+                    select(models.DatasetExample)
+                    .join(
+                        models.DatasetExampleRevision,
+                        onclause=models.DatasetExampleRevision.dataset_example_id
+                        == models.DatasetExample.id,
+                    )
+                    .where(
+                        and_(
+                            models.DatasetExample.id == example_id,
+                            models.DatasetExampleRevision.id == latest_revision_id,
+                            models.DatasetExampleRevision.revision_kind != "DELETE",
```

Comment:
What if users want to find deleted examples?

Reply:
We don't currently allow that. We'll need to revisit if that is necessary.